swift-sdk/Internal/AnonymousUserMerge.swift

```diff
@@ -21,12 +21,12 @@ class AnonymousUserMerge: AnonymousUserMergeProtocol {
         self.anonymousUserManager = anonymousUserManager
     }
     
-    public func tryMergeUser(sourceUserId: String?, sourceEmail: String?, destinationUserIdOrEmail: String?, isEmail: Bool, merge: Bool, onMergeResult: @escaping MergeActionHandler) {
-        if (sourceUserId != nil && destinationUserIdOrEmail != nil && merge) {
-            let destinationEmail = isEmail ? destinationUserIdOrEmail : nil
-            let destinationUserId = isEmail ? nil : destinationUserIdOrEmail
+    func tryMergeUser(anonymousUserId: String?, destinationUser: String?, isEmail: Bool, shouldMerge: Bool, onMergeResult: @escaping MergeActionHandler) {
+        if (anonymousUserId != nil && destinationUser != nil && shouldMerge) {
+            let destinationEmail = isEmail ? destinationUser : nil
+            let destinationUserId = isEmail ? nil : destinationUser
             
-            apiClient.mergeUser(sourceEmail: sourceEmail, sourceUserId: sourceUserId,  destinationEmail : destinationEmail, destinationUserId: destinationUserId).onSuccess {_ in
+            apiClient.mergeUser(sourceEmail: nil, sourceUserId: anonymousUserId,  destinationEmail: destinationEmail, destinationUserId: destinationUserId).onSuccess {_ in
```

Comment:
Also now that we know sourceEmail will always be nil, do we need to pass in anonymousUserID explicitly? SDK should know internally what userID it has.

Reply:
Gotcha. Updated so that the local storage is referenced in the AnonymousUserMerge class.